src/dialog-controller.js

```diff
@@ -8,8 +8,10 @@ import {invokeLifecycle} from './lifecycle';
 export class DialogController {
   settings: any;
   constructor(renderer: DialogRenderer, settings: any, resolve: Function, reject: Function) {
+    let defaultSettings = renderer ? renderer.defaultSettings || {} : {};
```

Comment:
I'm going to merge and then take a look at the whole codebase and see if there's something that can be improved here. Thanks!